src/EditorFeatures/Core/Implementation/ExtractInterface/AbstractExtractInterfaceCommandHandler.cs

```diff
@@ -50,33 +50,37 @@ public bool ExecuteCommand(ExtractInterfaceCommandArgs args, CommandExecutionCon
         {
             using (context.OperationContext.AddScope(allowCancellation: true, EditorFeaturesResources.Extract_Interface))
             {
-                var document = args.SubjectBuffer.CurrentSnapshot.GetFullyLoadedOpenDocumentInCurrentContextWithChangesAsync(
-                    context.OperationContext).WaitAndGetResult(context.OperationContext.UserCancellationToken);
-                if (document == null)
+                var subjectBuffer = args.SubjectBuffer;
+                if (!Workspace.TryGetWorkspace(subjectBuffer.AsTextContainer(), out var workspace))
```

Comment:
given how often this is written should we consider just having an extension on ITextBuffer like .TryGetOwningWorkspace?